core/tests/protractor_utils/TopicsAndSkillsDashboardPage.js

```diff
@@ -414,6 +410,12 @@ var TopicsAndSkillsDashboardPage = function() {
       unassignSkillButon,
       'Unassign Skill button takes too long to be clickable');
     await unassignSkillButon.click();
+
+    await waitFor.modalPopupToAppear();
```

Comment:
Are the `assignedTopicNamesInput` items in the modal? If so, then what's the point of waiting for the modal to appear before waiting for the elements in the modal to appear?

Reply:
Hmm, For this specific case, we fetch the "modal data" from the backend, so it might not occur as soon as the modal has been opened. So to have a linear waiting process, this does makes sense right.

Modal is opened -> Backend call is made -> Then modal content get populated.
Does this makes sense @U8NWXD ?

Reply:
I think having both waitFor statements is fine, though I still think the first is unnecessary. Since we don't do anything in-between the two waits, the first waitFor would only be needed if the modal opening and the modal elements loading each took 5-10 seconds so that the total wait would be longer than the default timeout of 10 seconds. Still, having both waitFor calls might be helpful for clarity, so I think this is fine.